apps/backend/src/lib/base-outbox-listener.ts

```diff
@@ -0,0 +1,302 @@
+      (err) =>
+        logger.error(
+          { err, listenerId: this.listenerId },
+          "OutboxListener debouncer error",
+        ),
+    )
+
+    await this.setupListener()
+    this.startFallbackPoll()
+
+    logger.info(
+      {
+        listenerId: this.listenerId,
+        debounceMs: this.debounceMs,
+        maxWaitMs: this.maxWaitMs,
+        fallbackPollMs: this.fallbackPollMs,
+      },
+      "OutboxListener started",
+    )
+  }
+
+  async stop(): Promise<void> {
+    this.running = false
+
+    if (this.debouncer) {
+      this.debouncer.cancel()
+      this.debouncer = null
+    }
+
+    if (this.fallbackTimer) {
+      clearTimeout(this.fallbackTimer)
+      this.fallbackTimer = null
+    }
+
+    if (this.listenClient) {
+      try {
+        await this.listenClient.query(`UNLISTEN ${OUTBOX_CHANNEL}`)
+        this.listenClient.release()
+      } catch {
+        // Ignore errors during cleanup
+      }
+      this.listenClient = null
+    }
+
+    logger.info({ listenerId: this.listenerId }, "OutboxListener stopped")
+  }
+
+  private async setupListener(): Promise<void> {
+    try {
+      this.listenClient = await this.pool.connect()
+
+      this.listenClient.on("notification", () => {
+        if (this.debouncer) {
+          this.debouncer.trigger()
+        }
+      })
+
+      this.listenClient.on("error", (err) => {
+        logger.error(
+          { err, listenerId: this.listenerId },
+          "LISTEN client error, reconnecting...",
+        )
+        this.reconnectListener()
+      })
+
+      await this.listenClient.query(`LISTEN ${OUTBOX_CHANNEL}`)
+      logger.debug({ listenerId: this.listenerId }, "LISTEN connection established")
+    } catch (err) {
+      logger.error(
+        { err, listenerId: this.listenerId },
+        "Failed to setup LISTEN connection",
+      )
+      this.scheduleReconnect()
+    }
+  }
+
+  private reconnectListener(): void {
+    if (this.listenClient) {
+      try {
+        this.listenClient.release()
+      } catch {
+        // Ignore
+      }
+      this.listenClient = null
+    }
+    this.scheduleReconnect()
+  }
+
+  private scheduleReconnect(): void {
+    if (!this.running) return
+
+    setTimeout(() => {
+      if (this.running) {
+        this.setupListener()
+      }
+    }, 1000)
+  }
+
+  private startFallbackPoll(): void {
+    if (!this.running) return
+
+    this.fallbackTimer = setTimeout(async () => {
+      try {
+        await this.processEvents()
+      } catch (err) {
+        logger.error(
+          { err, listenerId: this.listenerId },
+          "OutboxListener fallback poll error",
+        )
+      }
+      this.startFallbackPoll()
+    }, this.fallbackPollMs)
+  }
+
+  private async processEvents(): Promise<void> {
+    if (!this.running) return
+
+    await withTransaction(this.pool, async (client) => {
+      // Check if we're in retry backoff
+      const isReady = await OutboxListenerRepository.isReadyToProcess(
```

Comment:
Are we relying on the polling making sure we pick up first after the next time to process?